scripts/ag

```diff
@@ -105,32 +147,118 @@ def clean_test():
-    is_test_db = cur.fetchone()[0].lower()
+    cur.execute('SET SEARCH_PATH TO ag, public')
+
+    try:
+        cur.execute('SELECT test_environment FROM settings')
+        is_test_db = cur.fetchone()[0].lower()
+    except ProgrammingError:
+        if not force:
+            raise
+
     cur.close()
     conn.close()
 
+    if not force and is_test_db != 'true':
+        raise OperationalError("The settings table indicates this is not "
+                               "a test database; aborting.")
+
     conn = connect(user=AMGUT_CONFIG.user, password=AMGUT_CONFIG.password,
-                   host=AMGUT_CONFIG.host, port=AMGUT_CONFIG.port)
+                   host=AMGUT_CONFIG.host, port=AMGUT_CONFIG.port,
+                   database=AMGUT_CONFIG.database)
     cur = conn.cursor()
 
+    cur.execute('SET SEARCH_PATH TO ag, public')
+
     conn.set_isolation_level(ISOLATION_LEVEL_AUTOCOMMIT)
 
-    if is_test_db == 'false':
-        raise OperationalError("The settings table indicates this is not "
-                               "a test database; aborting.")
-    else:
-        cur.execute('drop database {}'.format(AMGUT_CONFIG.database))
-        print("Database %s dropped" % AMGUT_CONFIG.database)
-        cur.execute('drop role ag_wwwuser')
-        print("role ag_wwwuser dropped")
+    click.echo("Dropping user ag_wwwuser")
+
+    try:
+        click.echo('Reassigning owned objects')
+        cur.execute('REASSIGN OWNED BY ag_wwwuser TO {}'
+                    .format(AMGUT_CONFIG.user))
+
+        # Database
+        click.echo('Revoking all privileges on {}'
+                   .format(AMGUT_CONFIG.database))
+        cur.execute('REVOKE ALL PRIVILEGES ON DATABASE {} '
+                    'FROM ag_wwwuser CASCADE'.format(AMGUT_CONFIG.database))
+
+        # public schema
+        click.echo('Revoking all privileges in schema public')
+        cur.execute('REVOKE ALL PRIVILEGES ON SCHEMA public FROM ag_wwwuser '
+                    'CASCADE')
+
+        click.echo('Revoking all privileges on all tables in public')
+        cur.execute('REVOKE ALL PRIVILEGES ON ALL TABLES IN SCHEMA public '
+                    'FROM ag_wwwuser CASCADE')
+
+        click.echo('Revoking all privileges on all functions in public')
+        cur.execute('REVOKE ALL PRIVILEGES ON ALL FUNCTIONS IN SCHEMA public '
+                    'FROM ag_wwwuser CASCADE')
+
+        # ag schema
+        click.echo('Revoking all privileges in schema ag')
+        cur.execute('REVOKE ALL PRIVILEGES ON SCHEMA ag FROM ag_wwwuser '
+                    'CASCADE')
+
+        click.echo('Revoking all privileges on all tables in ag')
+        cur.execute('REVOKE ALL PRIVILEGES ON ALL TABLES IN SCHEMA ag '
+                    'FROM ag_wwwuser CASCADE')
+
+        click.echo('Revoking all privileges on all functions in ag')
+        cur.execute('REVOKE ALL PRIVILEGES ON ALL FUNCTIONS IN SCHEMA ag '
+                    'FROM ag_wwwuser CASCADE')
+    except:
```

Comment:
True, probably better to make these into functions and loop through them, `pass`ing on each--but seems OK to leave for now.